### PR TITLE
Fix rangebreaks on heatmap with 2d z array

### PR DESCRIPTION
**Pull request by archmoj**
Fixes: #4818 | demo: [before](https://codepen.io/MojtabaSamimi/pen/BaoraoP?editors=0010) vs [after](https://codepen.io/MojtabaSamimi/pen/qBOoZVB?editors=1000).

@plotly/plotly_js

### CODE REVIEW

File: test/image/mocks/axes_breaks-heatmap1d.json [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                "2020-01-02 18:00",
+                "2020-01-02 18:10",
+                "2020-01-02 17:00",
+                "2020-01-02 17:10",
+                "2020-01-02 17:20",
+                "2020-01-02 17:30",
+                "2020-01-02 17:40",
+                "2020-01-02 17:50",
+                "2020-01-02 18:00",
+                "2020-01-02 18:10"
```
Comment (alexcjohnson) [inline test/image/mocks/axes_breaks-heatmap1d.json:69]: These tests include some removed data, which is good - but can we also include some data on the other side of the break, ie the next morning?

**Comment by alexcjohnson** [comment]
Does this fix also support `histogram2d` with rangebreaks?

File: test/image/mocks/axes_breaks-heatmap1d.json (hunk 1 above)
Comment (archmoj) [reply test/image/mocks/axes_breaks-heatmap1d.json]: Good call. Done in https://github.com/plotly/plotly.js/pull/4821/commits/e2cf395aeec7ea1c2950a7579d157671f1c85fdb.

**Comment by archmoj** [comment]
> Does this fix also support `histogram2d` with rangebreaks?

Not sure about that one. Is there any problem you noticed with `histogram2d` and `rangebreaks`?

**Comment by alexcjohnson** [comment]
I haven't tried `histogram2d` with `rangebreaks` but I know it shares a lot of code with `heatmap` so seems natural to address them together.

**Comment by archmoj** [comment]
I added tests for `contour` in https://github.com/plotly/plotly.js/pull/4821/commits/a7676bd2798c335add2abf052e230572eba69a35.
The `histogram2d` case is tracked in #4825 amd it would be addressed in a separate PR.

**Pull request merged**